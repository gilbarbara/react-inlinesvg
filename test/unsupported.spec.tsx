import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import ReactInlineSVG from '../src';

declare let window: any;

const mockOnError = jest.fn();
let mockCanUseDOM = false;
let mockIsSupportedEnvironment = true;

jest.mock('../src/helpers', () => {
  const utils = jest.requireActual('../src/helpers');

  return {
    ...utils,
    canUseDOM: () => mockCanUseDOM,
    isSupportedEnvironment: () => mockIsSupportedEnvironment,
  };
});

const Loader = () => <div id="loader" />;

function setup(): Promise<ReactWrapper> {
  return new Promise((resolve) => {
    const wrapper = mount(
      <ReactInlineSVG
        src="http://localhost:1337/play.svg"
        onLoad={() => {
          setTimeout(() => {
            resolve(wrapper);
          }, 0);
        }}
        onError={(error) => {
          mockOnError(error);
          setTimeout(() => resolve(wrapper), 0);
        }}
        loader={<Loader />}
      />,
    );

    return wrapper;
  });
}
describe('unsupported environments', () => {
  it("shouldn't break without DOM ", async () => {
    const mockOnLoad = jest.fn();

    const wrapper = mount(
      <ReactInlineSVG
        src="http://localhost:1337/play.svg"
        onLoad={mockOnLoad}
        onError={mockOnError}
        loader={<Loader />}
      />,
    );

    expect(mockOnLoad).not.toHaveBeenCalled();
    expect(mockOnError).not.toHaveBeenCalled();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ src: 'http://localhost:1337/player.svg' });
  });

  it('should warn the user if fetch is missing', async () => {
    const globalFetch = fetch;
    window.fetch = undefined;

    mockCanUseDOM = true;
    mockIsSupportedEnvironment = true;

    const wrapper = await setup();

    expect(mockOnError).toHaveBeenCalledWith(new Error('fetch is not a function'));
    expect(wrapper.html()).toMatchSnapshot();

    window.fetch = globalFetch;
  });

  it("shouldn't not render anything if is an unsupported browser", async () => {
    mockCanUseDOM = true;
    mockIsSupportedEnvironment = false;

    const wrapper = await setup();

    expect(mockOnError).toHaveBeenCalledWith(new Error('Browser does not support SVG'));
    expect(wrapper.html()).toMatchSnapshot();
  });
});
