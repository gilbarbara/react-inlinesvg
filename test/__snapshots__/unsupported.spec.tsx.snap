// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unsupported environments should warn the user if fetch is missing 1`] = `null`;

exports[`unsupported environments shouldn't break without DOM  1`] = `
<div id="loader">
</div>
`;

exports[`unsupported environments shouldn't not render anything if is an unsupported browser 1`] = `null`;
